src/core/ext/filters/client_channel/resolver/dns/c_ares/grpc_ares_wrapper.cc

```diff
@@ -1166,11 +1164,11 @@ static void grpc_resolve_address_ares_impl(const char* name,
   r->on_resolve_address_done = on_done;
   GRPC_CLOSURE_INIT(&r->on_dns_lookup_done, on_dns_lookup_done, r,
                     grpc_schedule_on_exec_ctx);
-  r->ares_request = grpc_dns_lookup_ares(
+  r->ares_request = std::unique_ptr<grpc_ares_request>(grpc_dns_lookup_ares(
```

Comment:
This should be:

```
r->ares_request = absl::make_unique<grpc_ares_request>(...);
```

Reply:
`make_unique` doesn't work here since we only have the default constructor for `grpc_ares_request`, so it can't do the argument forwarding

Reply:
Whoops -- good point.